scipy/_lib/tests/test__util.py

```diff
@@ -243,3 +246,19 @@ def test_rng_integers():
     assert np.max(arr) == 4
     assert np.min(arr) == 0
     assert arr.shape == (100, )
+
+
+class TestValidateInt:
+
+    def test_validate_int(self):
+        n = _validate_int(4, 'n')
+        assert n == 4
```

Comment:
I pseudo expected `operator.index(Fraction(2, 1))` to work maybe also test `operator.index(np.array(1))`

Reply:
The test looks at the *type* of the object, not the value.  In general, not all `Fraction` objects are integers, just like not all `float` objects are integers.  I added a few more tests, including one to test that a scalar array is acceptable, and another that a `Fraction` is not (even if the value of the `Fraction` is actually an integer).